publisher/publisher.go

```diff
@@ -369,3 +391,37 @@ func CreateTestingSignedSCT(req []string, k *ecdsa.PrivateKey, precert bool, tim
 	jsonSCT, _ := json.Marshal(jsonSCTObj)
 	return jsonSCT
 }
+
+func (pub *Impl) ProbeLogs() {
+	wg := new(sync.WaitGroup)
+	for _, log := range pub.ctLogsCache.LogURIs() {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			c := http.Client{
+				Timeout: time.Minute*2 + time.Second*30,
+			}
+			url, err := url.Parse(log)
+			if err != nil {
+				pub.log.Errf("failed to parse log URI: %s", err)
+			}
+			url.Path = "/ct/v1/get-sth"
+			s := time.Now()
+			resp, err := c.Get(url.String())
+			took := time.Since(s).Seconds()
+			var success bool
+			var status string
+			if err == nil {
+				defer resp.Body.Close()
+				success = resp.StatusCode == http.StatusOK
+				status = resp.Status
+			}
+			pub.metrics.probeLatency.With(prometheus.Labels{
+				"log":     log,
+				"success": fmt.Sprintf("%t", success),
```

Comment:
Do you think the `success` label is necessary or should we match `http.StatusOK` in Prometheus queries instead of increasing the cardinality of the label set on this metric?